client/my-sites/stats/stats-post-anniversaries/index.jsx

```diff
@@ -0,0 +1,209 @@
+
+	const allPosts = Array.prototype.concat( ...postsByYear );
+
+	const cardClasses = classNames( 'stats-module', 'is-expanded', 'summary', {
+		'is-loading': requesting,
+		'has-no-data': ! allPosts.length,
+		'is-showing-error': ! allPosts.length,
+	} );
+
+	return (
+		<div>
+			<SectionHeader
+				label={ translate( 'Anniversaries', {
+					context: 'Stats: title of module',
+				} ) }
+			/>
+			<Card className={ cardClasses }>
+				{ siteId && <QueryPosts siteId={ siteId } query={ oldestPostQuery } /> }
+				{ siteId &&
+					postsByYearQueries.map(
+						( query, i ) => <QueryPosts key={ i } siteId={ siteId } query={ query } />,
+					) }
+
+				<StatsModulePlaceholder isLoading={ requesting } />
+
+				{ ! requesting &&
+					allPosts &&
+					( allPosts.length === 0
+						? <EmptyMessage />
+						: <PostsList allPosts={ allPosts } postsByYear={ postsByYear } /> ) }
+			</Card>
+		</div>
+	);
+};
+
+const CalendarIcon = () => (
+	<span style={ { position: 'relative', top: '-2px' } }>
+		<Gridicon icon="calendar" size={ 18 } />
+	</span>
+);
+
+const PostsList = ( { allPosts, postsByYear } ) =>
+	( allPosts.length === 1
+		? <SinglePost post={ allPosts[ 0 ] } />
+		: <GroupedPosts postsByYear={ postsByYear } /> );
+
+const SinglePost = localize( ( { translate, post } ) => {
+	const yearsAgo = moment()
+		.startOf( 'year' )
+		.diff( moment( post.date ).startOf( 'year' ), 'years' );
+	return (
+		<StatsContentText>
+			<p>
+				<CalendarIcon />
+				{ translate(
+					'%(yearsAgo)d year ago on this day, {{href}}%(title)s{{/href}} was published.',
+					'%(yearsAgo)d years ago on this day, {{href}}%(title)s{{/href}} was published.',
+					{
+						count: yearsAgo,
+						args: { yearsAgo, title: post.title },
+						components: {
+							href: <a href={ post.URL } target="_blank" rel="noopener noreferrer" />,
+						},
+						context: 'Stats: Sentence showing what post was published some years ago',
+					},
+				) }
+			</p>
+		</StatsContentText>
+	);
+} );
+
+const GroupedPosts = localize( ( { translate, postsByYear } ) => (
+	<div>
+		<StatsContentText>
+			<p>
+				{ translate( 'Published on this day in the past years:', {
+					context: 'Stats: Sentence preceding the list of posts published in the previous years',
+				} ) }
+			</p>
+		</StatsContentText>
+		<StatsListLegend label={ translate( 'Post' ) } value={ translate( 'Year' ) } />
+		<StatsList
+			moduleName={ 'postAnniversaries' }
+			data={ postsByYear.reduce(
+				( allPosts, posts, i ) =>
+					allPosts.concat(
+						posts.map( post => ( {
+							link: post.URL,
+							label: post.title,
+							labelIcon: 'calendar',
+							value: {
+								type: 'raw',
+								value: moment().subtract( i + 1, 'years' ).format( 'YYYY' ),
+							},
+						} ) ),
+					),
+				[],
+			) }
+		/>
+
+	</div>
+) );
+
+const EmptyMessage = localize( ( { translate } ) => (
+	<ErrorPanel
+		message={ translate( 'No anniversaries today!', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?